antiscum updates

killer2k1 opened the pull request.

killer2k1 committed:
antiscum onKilled function fix

Client::onKilled --> AntiScum::onKilled

killer2k1 committed:
client onkilled to antiscum on killed

killer2k1 committed:
createServer function

apparently this needs to be executed before other .cs files

killer2k1 committed:
cancel half countdown if capout occurs

in the event that a capout occurs in the first half, if this happens under 30 seconds when the notifications countdown has started its process, both countdowns will be activated, this update cancels out the notifications countdown so that only the countdown for the second half remains

killer2k1 committed:
suicide time checker update

- made it so suicide time checker does not count observers into the equation.
- changed multiplier value to 0.6 so that time checks every 7 suicides now instead of every 8 suicides.

killer2k1 committed:
observer spawn glitch

- fixed glitch where if you're admin and you join observer while the match has started, it spawns you into thin air
- also fixed bug where if you are on a team while in tourney mode and you select observer it just respawns you
- made it so Automatic team selection option is removed in tourney mode

killer2k1 committed:
Merge branch 'tribesonecommunity:master' into master

killer2k1 committed:
antiscum random messages

added some more fun/playful messages when players get punished by anti-scum

killer2k1 committed:
more precise 30 sec check

at the 40 second mark, schedule another check in 10 seconds, then display the message directly at 30 second mark

killer2k1 committed:
more fixes with 2min warning

in the event a pause occurs within the 3 to 2 minute mark, and the 2 minute warning has been scheduled, this now resets the timer to check again for the 2 minute warning if the paused time exceeds that of the scheduled check time

killer2k1 committed:
Merge branch 'master' into master

killer2k1 committed:
Pause / OOB quality fix update

- made it so when flag carrier exits the mission area, a pause cannot occur, once user re-enters mission area, pause enabled again
- this is to prevent the rare case of pausing the game when someone was OOB with flag (with intention of either throwing it out or bringing it back in)
- if someone pauses the game while someone is OOB with flag, itll be returned by the function of the game.

- fixed small issue in FFA mode where if balance mode was disabled would still recognize the end of the match as the "First half".

killer2k1 committed:
tab to space for code.server.cs

killer2k1 committed:
anti-scum update

- This update prevents flag carriers from self-tossing at the 0-second mark and never taking damage. Grabbing the flag at the 0 second mark will now apply 20% damage upon grab.

killer2k1 committed:
Delete code.antiscum.cs

killer2k1 committed:
anti-scum update

- This solves the issue of self-tossing at the 0-second marker and taking no damage. From now on if the timer reaches 0 seconds, flag carrier that attempts to continue to self-toss will inflict 20% damage upon each grab.

killer2k1 committed:
Various fixes - MA / Dmg Tracking Added

killer2k1 committed:
damage tracking - prevent client side hit sounds

also removed global message when player with flag enters / exits OOB

dwayneyuen committed:
Merge branch 'master' into master

killer2k1 committed:
antiscum updates

-bring back initial message
-fix double beep sound
-global teamkilled variables set to local

dwayneyuen commented on base/scripts/code.ratings.cs:
```diff
@@ -1,32 +1,32 @@
+$Stats::Rating["NadeJump"] = 0;
+$Stats::Rating["MidAir"] = 5;
+$Stats::Rating["MidAirLong"] = 10;
+$Stats::Rating["MidAirCatch"] = -5;
```
i think this is too much. at -5, getting midaired is 2.5x worse than dying!

dwayneyuen commented on base/scripts/freeze.cs:
```diff
@@ -1,248 +1,214 @@
-// Module for Tribes servers used to pause/resume game. Written as part of FSTAT v2.x, here modified for standalone purposes.
-// By Slitz
-// --------------------------------------------------------------------------------------------------------------------------
-// ** Install notes **
```
could you put these notes back? there's useful information here.

dwayneyuen commented on base/scripts/objectives.cs:
```diff
@@ -273,16 +283,16 @@
 function Game::UpdateTimeOnly() {
     
-    %curTimeLeft = ($Server::timeLimit * 60) + $missionStartTime - $simTimeTemp;
+    %curTimeLeft = ($Server::timeLimit * 60) + ($missionStartTime - $simTimeTemp);
```
just noting, nothing to change - this doesn't do anything

5 + (4 - 3) is the same as 5 + 4 - 3

killer2k1 replied:
true but my life is more at ease when things are in parenthesis.

dwayneyuen commented on base/scripts/objectives.cs:
```diff
@@ -466,8 +476,27 @@
  %set = nameToID("MissionCleanup/ObjectivesSet");
  %this.outArea = "";
+ 
+ %playerClient = Player::getClient(%this);
+ %playerTeam = GameBase::getTeam(%this);
+ %ClientName = Client::getName(%playerClient);
+    
+    if (%playerTeam == 0) {
+        %flagTeam = 1;
+    }
+    else {
+       %flagTeam = 0; 
+    }
```
this is more concise and also looks like you don't use ClientName anymore:
```suggestion
 %playerClient = Player::getClient(%this);
 %playerTeam = GameBase::getTeam(%this);
 %flagTeam = (%playerTeam + 1) % 2;
```

dwayneyuen commented on base/scripts/objectives.cs:
```diff
@@ -466,8 +476,27 @@
+    
  %set = nameToID("MissionCleanup/ObjectivesSet");
  %this.outArea = "";
+ 
+ %playerClient = Player::getClient(%this);
+ %playerTeam = GameBase::getTeam(%this);
+ %ClientName = Client::getName(%playerClient);
+    
+    if (%playerTeam == 0) {
+        %flagTeam = 1;
+    }
+    else {
+       %flagTeam = 0; 
+    }
+    
+ if((%playerClient == $freeze::FlagClient[%flagTeam]) && (!$TwoMinWarning)) {
+     
+    $freeze::OOB[%flagTeam] = false;
+    //MessageAllExcept(%playerClient, 0, %ClientName @ " entered the mission area. [PAUSE ENABLED]");
+
+ }
```
slightly more concise, if you like it
```suggestion
 %playerClient = Player::getClient(%this);
 %playerTeam = GameBase::getTeam(%this);
 %flagTeam = (%playerTeam + 1) % 2;

 if((%playerClient == $freeze::FlagClient[%flagTeam]) && (!$TwoMinWarning)) {
    $freeze::OOB[%flagTeam] = false;
 }
```

dwayneyuen commented on base/scripts/objectives.cs:
```diff
@@ -477,8 +506,25 @@
 
 function Player::leaveMissionArea(%this)
 {
+    %playerClient = Player::getClient(%this);
+    %ClientName = Client::getName(%playerClient);
+    %playerTeam = GameBase::getTeam(%this);
+    
+    if (%playerTeam == 0) {
+        %flagTeam = 1;
+    }
+    else {
+       %flagTeam = 0; 
+    }
+    // if we have the flag and we've entered OOB
+    if((%playerClient == $freeze::FlagClient[%flagTeam]) && (!$TwoMinWarning)) {
+        
+        $freeze::OOB[%flagTeam] = true;
+        //MessageAllExcept(%playerClient, 1, %ClientName @ " left the mission area. [PAUSE DISABLED]");
+    }
+
```
slightly more concise, if you like it
```suggestion
    %playerClient = Player::getClient(%this);
    %playerTeam = GameBase::getTeam(%this);
    %flagTeam = (%playerTeam + 1) % 2;
    // if we have the flag and we've entered OOB
    if((%playerClient == $freeze::FlagClient[%flagTeam]) && (!$TwoMinWarning)) { 
        $freeze::OOB[%flagTeam] = true;
    }
```

dwayneyuen reviewed (commented):
some misc minor comments.

killer2k1 committed:
some changes

midairlongcatch
freeze notes

killer2k1 committed:
cleaned up

-was producing double messages without the return in the event the %force flag was sent as well as one of the pre-defined times was active (30, 15, 10 etc)

dwayneyuen commented on base/scripts/code.antiscum.cs:
```diff
@@ -43,17 +43,15 @@
   if (%timeLeft > 5) {
-    if (%timeLeft == 30 || %timeLeft == 15 || %timeLeft == 10) {
+      
+    if (%timeLeft == 30 || %timeLeft == 20 || %timeLeft == 10 || %force) {
```
hey, this still isn't equivalent to before. the previous behavior would send the message as soon as someone left the safe zone. i think this is important for the player to know exactly when their timer begins. the current implementation doesn't tell them when they leave the safe zone.

is your goal here to avoid the double message? if so, you can put back the original if-statement block and change this logic to check for `!%force`. that way the two blocks can't both occur.

dwayneyuen reviewed (approved):
lgtm - thanks for the updates!
